src/goose/api/dead_jobs.clj

```diff
@@ -1,58 +1,50 @@
 (ns goose.api.dead-jobs
   (:require
-    [goose.brokers.broker :as broker]
-    [goose.brokers.redis.commands :as redis-cmds]
-    [goose.defaults :as d]
-    [goose.scheduler :as scheduler]))
+    [goose.brokers.broker :as b]))
 
 (defn size
   "Get size of Dead Jobs."
   [broker-opts]
-  (let [conn (broker/new broker-opts)]
-    (redis-cmds/sorted-set-size conn d/prefixed-dead-queue)))
+  (let [broker (b/new broker-opts)]
+    (b/dead-jobs-size broker)))
 
 (defn find-by-id
   "Find a Dead Job by ID."
   [broker-opts id]
```

Comment:
Is it not a little strange that these APIs all accept `broker-opts`, but not the broker itself?

Reply:
I think it's worthwhile to have these APIs accept the broker, rather than create a new broker every time they are called. It will probably result in efficiency improvements across broker implementations. It's also a much more common type of API, I think.

Reply:
All APIs accept `broker-opts`, and not a `broker` because users have constructed a `broker-opts` when interacting with client/worker.
`broker/new` should be kept hidden from the users else they'll be confused as to why not pass a broker directly to a client/worker.

Reply:
1. If broker/new is handled internally, for users that's one less function to worry about (OLFTWA :wink: )
2. Since client & worker both accept broker-opts, it'll be confusing if API requires a broker